umbraco/Umbraco-CMS
# Fix URL with culture

Fixes #11687

The issue is that in multilingual setups `Model.Url(culture)` might return nothing (#), depending on the casing of the culture.

The issue was that when we initially create the route for the cache in `ContentCache.GetRouteByIdInternal`, we have a check:

```
var hasDomains = _domainCache.GetAssignedWithCulture(culture, n.Id);
```

this method `GetAssignedWithCulture`, tries to find a domain with the culture in all the domains in the cache, however this check was done without ignoring casing, leading to the route being unable to be created, this PR fixes this by ignoring the casing in the check.

Additionally I've added a minor fix where we tried to match the culture to the default culture, again without ignoring casing. 


## Testing 

Same as the original issue: 

1. Set up a basic 2 language multi-lingual site, be aware that the culture should have a hypen, since it'll be all lowercase regardless, I used Canadian French like the original issue 
2. Create a very basic document type for a home page that varies by culture
3. Setup domains for each culture
4. Save and publish each culture
5. Create add the code below to a template
6. Ensure that all urls are rendered.

### Template snippet
```
<div>
    Cultures Loop:<br>
    @foreach (var (culture, infos) in Model.Cultures)
    {
<ul>
    <li>Culture: @culture.ToString()</li>
    <li>Model.Url(culture): @Model.Url(culture)</li>
    <li>Model.Url(culture): @Model.UrlSegment(culture)</li>
</ul>
    }
</div>
```
